.github/workflows/publish.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and publish
+
+on:
+  push:
+    branches-ignore:
+      - 'main'
+      - '*backport*'
+    tags:
+      - 'v*'
+      - '!*dev*'
+      - '!*pre*'
+      - '!*post*'
+  pull_request:
+    branches:
+      - 'main'
+#    paths:
+#      - '.github/workflows/publish.yml'
+  # Allow manual runs through the web UI
+  workflow_dispatch:
+  schedule:
+    #        ┌───────── minute (0 - 59)
+    #        │ ┌───────── hour (0 - 23)
+    #        │ │ ┌───────── day of the month (1 - 31)
+    #        │ │ │ ┌───────── month (1 - 12 or JAN-DEC)
+    #        │ │ │ │ ┌───────── day of the week (0 - 6 or SUN-SAT)
+    - cron: '0 3 * * *'  # Every day at 03:00 UTC
```

Comment:
Probably not! What schedule would you like, if any? It's taking over an hour to build. Probably because it needs to build scikit-image from source quite often.

Reply:
We would probably never schedule it?

Reply:
Okay. Currently it will run for an hour every time we merge a backport. Is that necessary? We could just test the build daily and trigger a manual build when we think it's necessary?

Reply:
Yeah a daily build and trigger it manually seems like a good idea.

Reply:
["Scheduled workflows run on the latest commit on the default or base branch."](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onschedule) which means we can't have scheduled jobs on the backport branches!!! (Confirmed by checking our asv workflow.) There must be some way to have a bot manually trigger a run with the GitHub API but that's out of the scope of this PR I think. I'll just remove the schedule for now.

Reply:
Oh we never want to run this on backport PRs.

Reply:
I'd say only run it weekly if we can get it reporting to the element channel?

Reply:
I mean run after _merging_ backport PRs?

Reply:
Oh, nah. I think only on cron and manually. On tag we want to release after that.

Reply:
The new `scheduled_builds.yml` workflow will hopefully take care of the cron.

Reply:
We currently build wheels on every merge to a backport branch.

I am a little concerned that making it weekly on backport branches means that someone doing a release has another step to check before tagging, which is "make sure a wheel build has passed on the last commit". I am *very* against anything which adds steps to the release pipeline that isn't automated.

I propose we maintain the status quo of building wheels on every merge to a backport branch (which aren't *that* frequent) and add a request to sunpy/sunpy-project#2 for more automation of bug fix releases. [I think I can probably make a whole bug fix release be "open an issue and talk to a bot" with actions.]

Reply:
Yep, that is what I think :+1: